qa/workunits/rbd/compare_mirror_image_alternate_primary.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+set -ex
+
+IMAGE=image
+RBD_MIRROR_MODE=snapshot
+MOUNT=/mnt/test
+WORKLOAD_TIMEOUT=5m
+
+. $(dirname $0)/rbd_mirror_helpers.sh
+
+launch_manual_msnaps() {
+  local cluster=$1
+  local pool=$2
+  local image=$3
+
+  for i in {1..30}; do
+    mirror_image_snapshot $cluster $pool $image
+    sleep 3s;
+  done
+}
+
+run_bench() {
+  local mountpt=$1
+  local timeout=$2
+
+  KERNEL_TAR_URL="https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.14.280.tar.gz"
+  sudo wget $KERNEL_TAR_URL -O $mountpt/kernel.tar.gz
+  sudo timeout $timeout bash -c "tar xvfz $mountpt/kernel.tar.gz -C $mountpt \
```

Comment:
I don't know if you have noticed, but "v" in "xvfz" here is incredibly important.  Omitting it would make the entire test bogus (always pass) because `pv` wouldn't have anything to rate limit and untar would complete in a seconds.  This would need a comment or, better, reformulating so that `pv` limits untar directly rather than rate limiting the listing of file names on stdout.

Reply:
Thanks for pointing this out. I missed it. Insted, as you suggested, I rate limited the untar.  Rate limiting the untar as follows, `zcat $mountpt/kernel.tar.gz | pv -L 256K --timer | tar xf - -C $mountpt`, took close to an hour in my local setup. The duration of the workload was similar to `tar xvfz $mountpt/kernel.tar.gz -C $mountpt | pv -L 1k --timer` that took approximately 50 minutes to complete.

Reply:
Did you try `zcat $mountpt/kernel.tar.gz | pv -L 256K --timer | tar xf - -C $mountpt` against the bad build?  Does it still reproduce the failure?

Reply:
Yes. See,
http://qa-proxy.ceph.com/teuthology/rraja-2024-02-08_03:12:35-rbd:mirror-wip-ajarr-61617-revert-corruption-fixes-distro-default-smithi/7551544/teuthology.log
http://qa-proxy.ceph.com/teuthology/rraja-2024-02-08_03:12:35-rbd:mirror-wip-ajarr-61617-revert-corruption-fixes-distro-default-smithi/7551548/teuthology.log
I can see the checksum mistmatch ('Mistmatch iteration at ') failure in both.